logstash-core/src/main/java/org/logstash/instrument/metrics/timer/TimerMetric.java

```diff
@@ -0,0 +1,44 @@
+package org.logstash.instrument.metrics.timer;
+
+import org.jruby.RubySymbol;
+import org.jruby.runtime.ThreadContext;
+import org.jruby.runtime.builtin.IRubyObject;
+import org.logstash.RubyUtil;
+import org.logstash.instrument.metrics.AbstractNamespacedMetricExt;
+import org.logstash.instrument.metrics.Metric;
+import org.logstash.instrument.metrics.MetricType;
+
+/**
+ * The {@code TimerMetric} is a logstash-internal extension of its public
+ * {@link co.elastic.logstash.api.TimerMetric} counterpart that provides read-
+ * and introspection-access via {@code Metric<Long>}.
+ */
+public interface TimerMetric extends co.elastic.logstash.api.TimerMetric, Metric<Long> {
+    Long getValue();
+
+    @Override
+    default MetricType getType() {
+        return MetricType.TIMER_LONG;
+    }
+
+    static TimerMetric create(final String name) {
+        // INTERNAL-ONLY system property escape hatch, set with `metric.timers` config in logstash.yml
+        switch (System.getProperty("ls.metric.timers", "delayed")) {
```

Comment:
What is the purpose of considering `ls.metric.timers`? For any input, we are returning `AfterCompletionTimerMetric`, removing confusing `switch/case` would be better if there is no any hidden logic.

Reply:
The meta-spec defines that we will provide a `metric.timers` config in the `logstash.yml` so that when we introduce live-tracking timers it can be selected.

This system property is set at startup with the value from the `metric.timer` setting, and is the injection point where we will add `live` timer routing. The PR as of when you reviewed it only had the one implementation so the switch statement is superfluous, but it is intended to make review of the additional implementation more straight-forward.